FHIR-Converter/client/src/test/suite/converter/engine/fhir-converter-engine.test.ts

```diff
@@ -0,0 +1,180 @@
+		try {
+			hl7v2Engine.process(emptyDataPath);
+			assert.strictEqual(true, false);
+		} catch (error) {
+			assert.strictEqual(true, true);
+		}
+	});
+});
+
+suite('CCDA Converter Engine Test Suite', () => {
+	const testPath = path.join(__dirname, '../../../../../../test-data');
+	const resultFolder = path.join(testPath, 'result');
+	const activeDataPath = path.join(testPath, 'data/C-CDA/CCD.ccda');
+	const activeDataXmlPath = path.join(testPath, 'data/C-CDA/C-CDA_R2-1_CCD.xml');
+	const invalidActiveDataPath = path.join(testPath, 'data/C-CDA/CCD-error.ccda');
+	const emptyDataPath = path.join(testPath, 'data/C-CDA/empty.ccda');
+	const templateFolder = path.join(testPath, 'templates/C-CDA');
+	const invalidTemplateFolder = path.join(testPath, 'templates');
+	const resultFile = path.join(resultFolder, 'temp.json');
+	const rootTemplate = 'CCD.liquid';
+	const invalidrootTemplate = 'Invalid_template';
+	const ccdaEngine = new FhirConverterEngine(templateFolder, rootTemplate, resultFolder);
+
+	beforeEach(() => {
+		if (fs.existsSync(resultFile)) {
+			fs.unlinkSync(resultFile);
+		}
+	});
+
+	test('Function constructor - should return a engine', async () => {
+		const engine = new FhirConverterEngine(templateFolder, rootTemplate, resultFolder);
+		assert.strictEqual(engine instanceof FhirConverterEngine, true);
+	});
+
+	test('Function process - should return a json object with OK status given data with .ccda, template and template folder', async () => {
+		assert.strictEqual(false, fs.existsSync(resultFile));
+		ccdaEngine.process(activeDataPath);
+		assert.strictEqual(true, fs.existsSync(resultFile));
+		const msg = JSON.parse(fs.readFileSync(resultFile).toString());
+		assert.strictEqual('OK', msg.Status);
+	}).timeout(20000);
+	
+	test('Function process - should return a json object with OK status given data with .xml, template and template folder', async () => {
+		assert.strictEqual(false, fs.existsSync(resultFile));
+		ccdaEngine.process(activeDataXmlPath);
+		assert.strictEqual(true, fs.existsSync(resultFile));
+		const msg = JSON.parse(fs.readFileSync(resultFile).toString());
+		assert.strictEqual('OK', msg.Status);
+	}).timeout(20000);
+	
+	test('Function process - should throw an error given invalid data', async () => {
+		try {
+			ccdaEngine.process(invalidActiveDataPath);
+			assert.strictEqual(true, false);
+		} catch (error) {
+			assert.strictEqual(true, true);
+		}
+	});
+	
+	test('Function process - should throw an error given invalid entry template', async () => {
+		try {
+			const ccdaEngineInvalidrootTemplate = new FhirConverterEngine(templateFolder, invalidrootTemplate, resultFolder);
+			ccdaEngineInvalidrootTemplate.process(activeDataPath);
+			assert.strictEqual(true, false);
+		} catch (error) {
+			assert.strictEqual(true, true);
+		}
+	});
+
+	test('Function process - should throw an error given invalid template folder', async () => {
```

Comment:
Just curious what will happen if users choose C-CDA data but HL7 v2 templates, or vice versa?

Reply:
The engine will throw the error "The input data could not be parsed correctly", so I will display this error message from engine